ToolBox/LeanDataReader.cs

```diff
@@ -48,6 +52,66 @@ public LeanDataReader(SubscriptionDataConfig config, Symbol symbol, Resolution r
             _config = config;
         }
 
+        /// <summary>
+        /// Initialize a instance of LeanDataReader from a path to a zipped data file.
+        /// It also supports declaring the zip entry CSV file for options and futures.  
+        /// </summary>
+        /// <param name="filepath">Absolute or relative path to a zipped data file, optionally the zip entry file can be declared by using '#' as separator.</param>
+        /// <example>
+        /// var dataReader = LeanDataReader("../relative/path/to/file.zip")
+        /// var dataReader = LeanDataReader("absolute/path/to/file.zip#zipEntry.csv")
+        /// </example>
+        public LeanDataReader(string filepath)
+        {
+            Symbol symbol;
+            DateTime date;
+            Resolution resolution;
+            string zipEntry = null;
+
+            var localDiskMapFileProvider = new LocalDiskMapFileProvider();
+            Composer.Instance.AddPart<IMapFileProvider>(localDiskMapFileProvider);
```

Comment:
I must have missed this in the previous review, but typically a constructor shouldn't be adding parts to the global composer instance. For example, if this constructor gets called twice we're adding two instances? It's best that this type of composer management happen at the application level and not at the component/ctor level.

Consider setting directly from python at the top of the script -- if required, can add static helper c# function to make it easier, but these two lines should be invokable via python directly.